external-builds/pytorch/patches/v2.7.0/third_party/fbgemm/hipified/0001-UtilsAvx512.cpp-fix-r-may-be-used-uninitialized-for-.patch

```diff
@@ -0,0 +1,30 @@
+From 48681e70549013664690c0c5376f1a43c06dd3aa Mon Sep 17 00:00:00 2001
+From: Scott Tsai <scottt.tw@gmail.com>
+Date: Sat, 22 Mar 2025 06:17:51 +0800
+Subject: [PATCH] UtilsAvx512.cpp: fix r may be used uninitialized for gcc14
+
+This version of fbgemm uses `-Werror` thus the warning breaks the build.
+This is also reported upstream by others as https://github.com/pytorch/pytorch/issues/129358
+
+ROCm/TheRock is building pytorch in Ubuntu 22.04 but Fedora 41 has a
+newer gcc and would run into this bug.
+---
+ src/UtilsAvx512.cc | 2 ++
+ 1 file changed, 2 insertions(+)
+
+diff --git a/src/UtilsAvx512.cc b/src/UtilsAvx512.cc
+index cf00613..9b874ee 100644
+--- a/src/UtilsAvx512.cc
++++ b/src/UtilsAvx512.cc
+@@ -921,6 +921,8 @@ static inline void transpose_contiguous_16x2_block(
+     int mrem = 16) {
+   __m512i r[2], d[2];
+   int i = 0;
++  r[0] = _mm512_setzero_si512();
++  r[1] = _mm512_setzero_si512();
```

Comment:
Do we want to zero out before initializing `i` like in upstream? https://github.com/q10/FBGEMM/commit/963fb151d00a0201678c027c702315d969ff3805#diff-f60f605607290836730866d7a1f05bdaedef0eb3fbc856b1155c801273f7b916

Reply:
@marbre , I've amended the patch to do this. https://github.com/ROCm/TheRock/compare/27f4825c90f1efeb9bdc4f21b77c6ee83a7c6047..4e0c40671f8902ab320f8f53f51fb5b37370927c